### DISCUSSION TITLE
Direction of current (isec ampa) in L5 pyramidal vs L2/3 pyramidal plotting

### DISCUSSION THREAD

**Discussion by mkhalil8** [jonescompneurolab/hnn-core#731] [neuro]
![Current_direction](https://github.com/jonescompneurolab/hnn-core/assets/14954705/5d6c3c2c-763f-411d-a890-94a266b45b55)

@rythorpe @ntolley 
Hi Ryan and Nick, 
I discussed with Ryan briefly this finding on plotting the current of AMPA in the apical tufts of L2/3 pyramidal and L5 pyramidal, one is negative and one is positive. What are your thoughts on that?

**Comment by ntolley** [comment]
@mkhalil8 it'd be useful to plot the membrane potential of the  soma and apical tuft for the same cell along with AMPA currents

The directionality of current flow is going to depend on the reversal potential of the AMPA receptor, so depending on when the cell spike the directionality will flip if the membrane potential is depolarized or hyperpolarized relative to the reversal potential

**Reply by mkhalil8** [reply]
Wow, so the soma firing will affect the current at the apical tuft. is that through the intra axial current?. I will plot them together and post it.

**Reply by mkhalil8** [reply]
quick question, so far the isec and vsec data are not saved on the drive with the cell_response object, correct?

**Reply by ntolley** [reply]
Yeah! It can be through intra axial currents (or NMDA/GABA receptors on the same sections). Basically anything that impacts the membrane potential will impact the currents.

I'd definitely recommend checking out a textbook/lecture on single cell electrophysiology to get a good intuition for these sorts measurements. For example I-V curves are a common tool for evaluating ion channels:

![image](https://github.com/jonescompneurolab/hnn-core/assets/55253912/37a00e70-e9dd-4379-9394-bbee6953d4ce)

https://en.wikipedia.org/wiki/Current%E2%80%93voltage_characteristic#In_electrophysiology

You can see that the flow of different ions goes from positive to negative depending on the membrane potential

**Reply by mkhalil8** [reply]
The current movement through the receptor (for example sodium through AMPA) is in the same direction every time, correct? because it is concentration vs voltage forces (nernst potential). But the status of the membrane (its voltage) is what will be affected by other factors like other receptors or soma firing.
I will read more about it as you suggested Nick. Thank you

**Reply by rythorpe** [reply]
Not quite. The current associated with a particular ion will change direction depending on the membrane potential in relation to the reversal potential for that current (i.e., the voltage force). So if the AMPA receptor has a reversal potential of 0 mV, then Na+ will flow outward (positive extracellular current) if the neuron is near the peak of an action potential (>0 mV), but inward (negative extracellular current) if the neuron is near its resting state (<0 mV).

**Comment by ntolley** [comment]
> quick question, so far the isec and vsec data are not saved on the drive with the cell_response object, correct?

Not sure I understand the question, are you referring to voltages/currents of the drive cells themselves?

**Reply by mkhalil8** [reply]
I mean the isec data of the cell response object:
`net.cell_response.isec[0][cell]['apical_tuft']['apical_tuft_ampa']`

**Reply by rythorpe** [reply]
If you recorded `vsec` during the simulation, you should be able to use a similar API to access the voltage values:
`net.cell_response.vsec`.

**Reply by mkhalil8** [reply]
Does it get saved on drive with the cell_response?

**Reply by rythorpe** [reply]
You mean do these values get saved via `cell_response.write('fname.txt')`? to `fname.txt`?

**Reply by mkhalil8** [reply]
Yes exactly!

**Reply by rythorpe** [reply]
Unfortunately no. This should be fixed soon, however, once we start saving `Network` and `CellResponse` objects within hdf5 files instead of relying on txt files.
